services/src/main/java/com/broadleafcommerce/subscriptionoperation/service/modification/AbstractGradeChangeSubscriptionModificationHandler.java

```diff
@@ -248,41 +325,58 @@ protected List<AddItemRequest> getDependentItems(
-                    AddItemRequest addOnRequest = getTypeFactory().get(AddItemRequest.class);
-                    if (itemChoice != null) {
-                        if (isItemARequiredAddOn(itemChoice)) {
-                            addOnRequest.setQuantity(itemChoice.getMinimumQuantity());
-                            Product defaultProduct = itemChoice.getDefaultProduct();
-                            if (defaultProduct != null) {
-                                addOnRequest.setProductId(defaultProduct.getId());
-                            }
-                            addOnRequest.setItemType(DefaultCartTypes.STANDARD.name());
-                            addOnRequest.setTermDurationType(
-                                    subscription.getTermDurationType());
-                            addOnRequest.setTermDurationLength(
-                                    subscription.getTermDurationLength());
-                            addOnRequest.getCartAttributes().put(SUBSCRIPTION_ACTION_FLOW,
-                                    DefaultSubscriptionActionFlow.UPGRADE.name());
-                            addOnRequest.getItemInternalAttributes()
-                                    .put(EXISTING_SUBSCRIPTION_ID, subscription.getId());
-                            return addOnRequest;
-                        }
-                    }
-                    return null;
-                })
-                .filter(Objects::nonNull)
+                .filter(option -> isItemChoiceProductOption(option.getType()))
+                .map(option -> buildAddItemRequestForAddOn(option, subscription))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Builds a {@link AddItemRequest} for any relevant add-ons. By default, this only creates
+     * requests for required add-ons.
+     * <p>
+     * In Broadleaf, an add-on is represented by a {@link ProductOption} of type
+     * {@link DefaultProductOptionType#ITEM_CHOICE}.
+     * 
+     * @param addOnOption A {@link ProductOption} of type
+     *        {@link DefaultProductOptionType#ITEM_CHOICE}.
+     * @param subscription The existing {@link Subscription}.
+     * 
+     * @return An {@link AddItemRequest} for the add-on.
+     *
+     * @see #populateDefaultProductAndOrVariantId(ItemChoice, AddItemRequest)
+     */
+    protected Optional<AddItemRequest> buildAddItemRequestForAddOn(
+            @lombok.NonNull ProductOption addOnOption,
+            @lombok.NonNull Subscription subscription) {
+        Assert.notNull(addOnOption.getItemChoice(), "Item Choice cannot be null.");
+
+        ItemChoice itemChoice = addOnOption.getItemChoice();
+
+        // todo try to match the existing options?
+        if (!isRequiredAddOn(itemChoice)) {
+            return Optional.empty();
+        }
+
+        AddItemRequest addOnRequest = getTypeFactory().get(AddItemRequest.class);
```

Comment:
```suggestion
        AddItemRequest addOnRequest = getTypeFactory().get(AddItemRequest.class);
        addOnRequest.setItemType(DefaultCartTypes.STANDARD.name());
```
Looks like the itemType was missed

Reply:
Standard is the default value, it does not need to be specified here.